pkg/util/k8sutil/k8sutil.go

```diff
@@ -81,6 +81,40 @@ const (
 
 var ErrDiscoveryTokenNotProvided = errors.New("cluster token not provided, you must provide a token when clustering mode is discovery")
 
+var CreateSvc CreateService = func(ctx context.Context, kubecli kubernetes.Interface, svcName string, clusterName string, ns string, ports []v1.ServicePort, owner metav1.OwnerReference, publishNotReadyAddresses bool, service *api.ServicePolicy, annotations map[string]string) error {
+	svc := newEtcdServiceManifest(svcName, clusterName, ports, publishNotReadyAddresses, annotations)
+
+    applyServicePolicy(svc, service)
+    addOwnerRefToObject(svc.GetObjectMeta(), owner)
+    _, err := kubecli.CoreV1().Services(ns).Create(ctx, svc, metav1.CreateOptions{})
+    if err != nil && !apierrors.IsAlreadyExists(err) {
+        return err
+    }
+	done := make(chan string)
+	go getServiceStatus(ctx, kubecli, svcName, ns, done)
+	status := <- done
+	if status != "created" {
+		return fmt.Errorf("failed to finish the service creation: %v", status)
+	}
+    return nil
+}
+
+func getServiceStatus(ctx context.Context, kubecli kubernetes.Interface, svcName string, ns string, status chan string) {
+	for i := 0; i < 20; i++ {
+		service, err := kubecli.CoreV1().Services(ns).Get(ctx, svcName, metav1.GetOptions{})
+		if err != nil {
+			status <- err.Error()
```

Comment:
If we receive an error fetching the service we could return the error probably, one thing is to have the status != created and other is having a problem IMO, in which case the kubecli.CoreV1().Services(ns).Get(ctx, svcName, metav1.GetOptions{}) could return an error?

Reply:
If the status != created then the call succeed and we have a service object, therefore error == nil, which won't be caught by the if in line 105.

Diving a bit deeper on this, possible errors are either if the service is not found or some kubernetes error (like unreachable or failed state) since it returns the straight call response: https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/service.go#L71